lgc/builder/BuilderRecorder.cpp

```diff
@@ -1979,6 +1966,23 @@ Instruction *BuilderRecorder::record(BuilderRecorder::Opcode opcode, Type *resul
     case Opcode::SubgroupSwizzleMask:
     case Opcode::SubgroupSwizzleQuad:
     case Opcode::SubgroupWriteInvocation:
+      // Subgroup operations.
+      func->addFnAttr(Attribute::ReadNone);
+      func->addFnAttr(Attribute::Convergent);
+      break;
+    case Opcode::Barrier:
+    case Opcode::DemoteToHelperInvocation:
+    case Opcode::EmitVertex:
+    case Opcode::EndPrimitive:
+    case Opcode::GetSubgroupSize:
```

Comment:
GetSubgroupSize actually returns a known wave size value. Could be readnone like normal ALU operations.

Reply:
Agreed.

Reply:
#775